LibVLCSharp.Forms/Shared/PlaybackControls.xaml.cs

```diff
@@ -0,0 +1,1384 @@
+        {
+            ((PlaybackControls)bindable).UpdateAudioTracksSelectionAvailability();
+        }
+
+        private static void IsClosedCaptionsSelectionButtonVisiblePropertyChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            ((PlaybackControls)bindable).UpdateClosedCaptionsTracksSelectionAvailability();
+        }
+
+        private static void IsPlayPauseButtonVisiblePropertyChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            ((PlaybackControls)bindable).UpdatePauseAvailability();
+        }
+
+        private static void IsCastButtonVisiblePropertyChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            ((PlaybackControls)bindable).UpdateCastAvailability();
+        }
+
+        private static void KeepScreenOnPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            ((PlaybackControls)bindable).UpdateKeepScreenOn((bool)newValue);
+        }
+
+        private static void LibVLCPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            var playbackControls = (PlaybackControls)bindable;
+            playbackControls.UpdateCastAvailability();
+            playbackControls.UpdateErrorMessage();
+        }
+
+        private static void MediaPlayerPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            ((PlaybackControls)bindable).OnMediaPlayerChanged((LibVLCSharp.Shared.MediaPlayer)oldValue, (LibVLCSharp.Shared.MediaPlayer)newValue);
+        }
+
+        private static void VideoViewPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            ((PlaybackControls)bindable).UpdateZoom();
+        }
+
+        private static void ZoomPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            ((PlaybackControls)bindable).UpdateZoom();
+        }
+
+        private void MediaPlayer_Buffering(object sender, MediaPlayerBufferingEventArgs e)
+        {
+            var value = (int)e.Cache / 100.0d;
+            if (BufferingProgress != value)
+            {
+                Device.BeginInvokeOnMainThread(() => BufferingProgress = value);
+            }
+        }
+
+        private void MediaPlayer_EncounteredError(object sender, EventArgs e)
+        {
+            UpdateState(VLCState.Error);
+        }
+
+        private void MediaPlayer_EndReached(object sender, EventArgs e)
+        {
+            UpdateState(VLCState.Ended);
+        }
+
+        private void MediaPlayer_LengthChanged(object sender, MediaPlayerLengthChangedEventArgs e)
+        {
+            UpdateTime(MediaPlayer?.Position);
+        }
+
+        private void MediaPlayer_MediaChanged(object sender, MediaPlayerMediaChangedEventArgs e)
+        {
+            Reset();
+        }
+
+        private void MediaPlayer_NothingSpecial(object sender, EventArgs e)
```

Comment:
I added it by security because I don't know when it can be called.

Reply:
This is an event handler, we can safely remove that (nothing will be called if no event handler, because we call it as `?.Invoke...`)

@mfkl : Do we really need to expose this event in the core?

Reply:
I mean it's not a dead event, it can be raised from libvlc. 

How relevant it is I guess that's debatable... but we can leave for completeness for now. Can reconsider for 4.0 I guess.